test/Table_Tests/src/Common_Table_Operations/Column_Operations_Spec.enso

```diff
@@ -779,6 +790,18 @@ spec setup =
                 x.round . to_vector . should_equal [1, -2, 4]
                 x.truncate . to_vector . should_equal [1, -2, 3]
 
+    Test.group prefix+"Date truncation" <|
+        Test.specify "should be able to truncate a column of Date_Times" pending=pending_datetime <|
+            tz = Time_Zone.parse "Europe/Warsaw"
+            dates_without_zone = [Date_Time.new 2020 10 24 1 2 3, Date_Time.new 2020 10 24 1 2 3]
+            dates_with_zone = dates_without_zone . map (d-> d.at_zone tz)
+            [dates_without_zone, dates_with_zone].map dates->
+                table = table_builder [["foo", dates]]
+                truncated = table.at "foo" . truncate
+                truncated . to_vector . should_equal [Date.new 2020 10 24, Date.new 2020 10 24]
+                truncated . value_type . should_equal Value_Type.Date
+                truncated.name . should_equal "truncate([foo])"
```

Comment:
I don't think this test works as you intend related to the timezone checking.

Please inspect the `table.at "foo" . value_type`.

I suspect that in Postgres, it will always be `Date_Time with_timezone=True` - that is because the way the tables are created using `table_builder` is we first create an in-memory table and upload it - but in-memory tables only support `Date_Time with_timezone=True` and not `Date_Time with_timezone=False`. If you want specific tests for `Date_Time with_timezone=True` you would need to use `connection.create_table` to be able to specify the types you want.

IMHO it is not necessary and _testing just one approach will suffice_, but ofc it is welcome to have better test coverage. But let's ensure the tests actually test what we think they do - because (unless I'm very mistaken), currently we have code paths that pretend to test the `Date_Time with_timezone=False` code path, but **they actually don't**.

Reply:
Also note that in Enso code, there is no such thing as date-time without timezone.

Your `dates_without_zone` actually **do have a timezone**. It is just the system default timezone.

This test is risky, because if my system default timezone is to the east from Warsaw, e.g. Tokyo, it will actually move one day backwards and this test will be failing. Our unit tests should work regardless of the location they are run in.

```
> d = Date_Time.new 2020 10 24 1 2 3 zone=(Time_Zone.parse "Asia/Tokyo")
>>> Nothing
> d.date
>>> 2020-10-24
> d.at_zone (Time_Zone.parse "Europe/Warsaw")
>>> 2020-10-23T18:02:03+02:00[Europe/Warsaw]
> d.at_zone (Time_Zone.parse "Europe/Warsaw") . date
>>> 2020-10-23
```

Reply:
> Please inspect the `table.at "foo" . value_type`.

I think the `value_type` should be checked in these tests to guarantee they are checking what we think they are checking.

Reply:
Done